src/main/java/client/ServiceManager.java

```diff
@@ -0,0 +1,153 @@
+package client;
+
+import commons.requests.*;
+import commons.responses.Response;
+import commons.rpc.Communicator;
+import commons.utils.Datetime;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class ServiceManager {
+    Communicator router;
+    InetAddress serverAddress;
+    int serverPort;
+
+    public ServiceManager(Communicator router, InetAddress serverAddress, int serverPort) {
+        this.router = router;
+        this.serverAddress = serverAddress;
+        this.serverPort = serverPort;
+    }
+    public void queryAvailability() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter facility name: ");
+        String facilityName = scanner.nextLine();
+        System.out.println("Please enter the start date [DD/MM/YYYY]: ");
+        String startDate = scanner.nextLine();
+        System.out.println("Please enter the end date [DD/MM/YYYY]: ");
+        String endDate = scanner.nextLine();
+
+        List<Datetime> dates = new ArrayList<>();
+        dates.addAll(getListOfDates(startDate, endDate));
```

Comment:
Just wanted to highlight that server/communicator module need to convert it to LocalDateTime object for server to manipulate (import java.time.*;)